### PR TITLE
Fix `istioctl proxy-config endpoint` command failed for ztunnel pods

### PR DESCRIPTION
**Pull request by kebe7jun**
**Please provide a description of this PR:**
`istioctl proxy-config endpoint` panic for ztunnel pods.
```
 kebe@Kebe-DaoCloud-PC  ~  /tmp/ik pc ep ztunnel-v84m6.istio-system
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x42ed85c]

goroutine 1 [running]:
istio.io/istio/istioctl/pkg/writer/envoy/clusters.retrieveEndpointAddress(...)
	/home/kebe/workspace/istio/istioctl/pkg/writer/envoy/clusters/clusters.go:73
istio.io/istio/istioctl/pkg/writer/envoy/clusters.(*ConfigWriter).PrintEndpointsSummary(0xc000213038, {{0x0, 0x0}, 0x0, {0x0, 0x0}, {0x0, 0x0}})
	/home/kebe/workspace/istio/istioctl/pkg/writer/envoy/clusters/clusters.go:125 +0x4fc
istio.io/istio/istioctl/cmd.endpointConfigCmd.func2(0xc0002e8f00?, {0xc0009d13d0?, 0x1?, 0x1?})
	/home/kebe/workspace/istio/istioctl/cmd/proxyconfig.go:935 +0x255
github.com/spf13/cobra.(*Command).execute(0xc0002e8f00, {0xc0009d13b0, 0x1, 0x1})
	/home/kebe/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:872 +0x694
github.com/spf13/cobra.(*Command).ExecuteC(0xc000eb1180)
	/home/kebe/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:990 +0x3bd
github.com/spf13/cobra.(*Command).Execute(...)
	/home/kebe/go/pkg/mod/github.com/spf13/cobra@v1.5.0/command.go:918
main.main()
	/home/kebe/workspace/istio/istioctl/cmd/istioctl/main.go:39 +0xdc
```

After:

```
 ✘ kebe@Kebe-DaoCloud-PC  ~  /tmp/ik pc ep ztunnel-v84m6.istio-system
ENDPOINT                                                STATUS      OUTLIER CHECK     CLUSTER
172.19.0.2:6443                                                                                            HEALTHY     OK                spiffe://cluster.local/ns/default/sa/bookinfo-reviews_to_https_kubernetes.default.svc.cluster.local_outbound_internal
172.19.0.2:6443                                                                                            HEALTHY     OK                spiffe://cluster.local/ns/default/sa/notsleep_to_https_kubernetes.default.svc.cluster.local_outbound_internal
172.19.0.2:6443                                                                                            HEALTHY     OK                spiffe://cluster.local/ns/default/sa/sleep_to_https_kubernetes.default.svc.cluster.local_outbound_internal
envoy://outbound_tunnel_lis_spiffe://cluster.local/ns/default/sa/bookinfo-details/10.244.0.13:9080         HEALTHY     OK                spiffe://cluster.local/ns/default/sa/bookinfo-details_to_http_details.default.svc.cluster.local_outbound_internal
...
```

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [X] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

File: istioctl/pkg/writer/envoy/clusters/clusters.go [hunk 1]
```diff
@@ -70,6 +70,12 @@ func retrieveEndpointAddress(host *adminapi.HostStatus) string {
 	if addr != nil {
 		return addr.Address
 	}
+	if addr := host.Address.GetEnvoyInternalAddress(); addr != nil {
```
Comment (howardjohn) [inline istioctl/pkg/writer/envoy/clusters/clusters.go:73]: can we follow the rest of the istioctl logic:

```
78:     if internal := addr.GetEnvoyInternalAddress(); internal != nil {
79-             switch an := internal.GetAddressNameSpecifier().(type) {
80-             case *core.EnvoyInternalAddress_ServerListenerName:
81-                     return fmt.Sprintf("envoy://%s/%s", an.ServerListenerName, internal.EndpointId)
82-             }
83-     }

```

Comment (howardjohn) [reply istioctl/pkg/writer/envoy/clusters/clusters.go:73]: which follows how envoy formats it

File: releasenotes/notes/fix-pc-ep-cmd-failed-for-ztunnel.yaml [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+# - installation
+# - istioctl
+# - documentation
+area: istioctl
+
+# releaseNotes is a markdown listing of any user facing changes. This will appear in the
+# release notes.
+releaseNotes:
+  - |-
+    **Fixed** an issue where `istioctl proxy-config endpoint` command failed for ztunnel listeners.
```
Comment (howardjohn) [inline releasenotes/notes/fix-pc-ep-cmd-failed-for-ztunnel.yaml:29]: No need for release notes on experimental branch

**Comment by kebe7jun** [comment]
@howardjohn Kind reminder.

**Review by howardjohn — APPROVED** [review]

**Pull request merged**